### PR TITLE
Fasih/ refactor event code

### PR DESCRIPTION
**Pull request by fasihali-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by hasan-deriv — APPROVED** [review]

**Review by aswathy-deriv — APPROVED** [review]

**Review by mayuran-deriv — APPROVED** [review]

**Review by konstantinosG-deriv — APPROVED** [review]

File: src/components/hooks/use-cookie-banner.ts [hunk 1]
```diff
@@ -38,10 +38,8 @@ export const useCookieBanner = () => {
             if (is_eu && !tracking_status) setShouldShow(true)
 
             if (allow_tracking) {
```
Comment (habib-deriv) [inline src/components/hooks/use-cookie-banner.ts:40]:
```suggestion
            if (allow_tracking) {
```
if (allowTracking && document.readyState === 'complete') {
    setGTMData({ event: 'allow_tracking' });
  }

**Review by NikitK-deriv — APPROVED** [review]
After Habib's note, approved

File: src/components/hooks/use-cookie-banner.ts (hunk 1 above)
Comment (fasihali-deriv) [reply src/components/hooks/use-cookie-banner.ts:40]: Ohhh yes I will update that @habib-deriv thanks for highlighting 🙇

**Review by habib-deriv — APPROVED** [review]
LGTM

**Pull request merged**